src/aiida_quantumespresso/tools/calculations/pw.py

```diff
@@ -92,3 +98,62 @@ def get_magnetic_configuration(self, atol: float = 0.5, ztol: float = 0.05) -> A
         return AttributeDict(
             {'structure': structure, 'magnetic_moments': None if non_magnetic else results['magnetic_moments']}
         )
+
+    def get_occupations(self, reshape=False) -> list[dict]:
+        """Return the occupations for a PwCalculation/PwBaseWorkChain node as a list of python dictionaries."""
+
+        # assert first that this is a Hubbard calculation
+        try:
+            type(self._node.inputs.structure) == HubbardStructureData
+        except exceptions.NotExistent as exc:
+            raise ValueError('The input structure is not a HubbardStructureData node.') from exc
+
+        
+        try: 
+            with self._node.outputs.retrieved.base.repository.open('data-file-schema.xml') as xml_file:
+                xml_parsed = ElementTree.parse(xml_file)
+
+                schema_filepath = get_schema_filepath(xml_parsed)
+                xsd = XMLSchema(schema_filepath)
+                xml_dictionary = xsd.to_dict(xml_parsed, validation='skip')
+
+                output_matrices = {}
+                for  ientry, atom_dict in enumerate(xml_dictionary['output']['dft']['dftU']['Hubbard_ns']):
+                    atom_specie = atom_dict['@specie']
+                    shell_label = atom_dict['@label']
+                    spin = 'up' if atom_dict['@spin'] == 1 else 'down'
+                    shell_dims = atom_dict['@dims']
+                    atom_index = atom_dict['@index']
+                    occ_matrix = np.array(atom_dict['$'])
+                    if reshape:
+                        occ_matrix = occ_matrix.reshape(shell_dims)
+
+                    atom_label = atom_index
+
+                    if atom_label not in output_matrices:
+                        output_matrices[atom_label] = {}
+                        output_matrices[atom_label]['specie'] = atom_specie
+                        output_matrices[atom_label]['shell'] = shell_label
+                        output_matrices[atom_label]['occupation_matrix'] = {'up': {}, 'down': {}}
+                    
+                    output_matrices[atom_label]['occupation_matrix'][spin] = occ_matrix
+        except Exception as exc:
+            # raise just a warning and return None
+            print(f'Warning: could not parse occupation matrices from XML file: {exc}')
+            return None
```

Comment:
as already written, we here stick with raising the error and not returning None, which is less informative